cpp/include/kvikio/defaults.hpp

```diff
@@ -27,7 +27,47 @@
 #include <kvikio/shim/cufile.hpp>
 
 namespace kvikio {
+/**
+ * @brief I/O compatibility mode.
+ */
+enum class CompatMode : uint8_t {
+  OFF,   // Enforce cuFile I/O. GDS will be activated if the system requirements for cuFile are met
+         // and cuFile is properly configured. However, if the system is not suited for cuFile, I/O
+         // operations under the OFF option may error out, crash or hang.
+  ON,    // Enforce POSIX I/O.
+  AUTO,  // Try cuFile I/O first, and fall back to POSIX I/O if the system requirements for cuFile
+         // are not met.
```

Comment:
I've updated the C++ source so that documentation on the enum is generated in Doxygen:
![image](https://github.com/user-attachments/assets/20e74588-f32d-452d-8f20-cf2f71bd18e8)
By "docstring" I guess you were referring to the Python counterpart? I haven't had success in getting the enum documentation generated in Sphinx. Perhaps this is because the enum is imported on the Cython side instead of explicitly defined in Python, and Sphinx does not capture that?

Reply:
By docstring, I was referring to in here:
```cpp
/**
 * @brief I/O compatibility mode.
 */
```

Reply:
> Perhaps this is because the enum is imported on the Cython side instead of explicitly defined in Python, and Sphinx does not capture that?

Yeah, I do a horrible hack in cudf to get documentation for these import enum numbers for pylibcudf:

https://github.com/rapidsai/cudf/blob/aa8c0c4a95920b75c09cf111df4e1c02752dade7/docs/cudf/source/conf.py#L662-L713

Reply:
Given @wence- 's approach, I've added the issue #555 for doc improvement.